### ISSUE TITLE
Add support for Lua 5.3 Integers

### ISSUE THREAD

**Issue by wilhelmy** [raboof/notion#139]
Lua 5.3 added Integers in addition to double precision floating point numbers (the previous default type for Lua numbers).

As a result, things which would be better represented by integers are still floating point numbers internally, e.g.

```
lua> ioncore.current():screen_of():mx_count()
6.0
```

**Comment by raboof** [comment]
I agree it would be much nicer to have integers rather than floats. On the other hand having them both in the same codebase sounds like a recipe for subtle bugs.

Perhaps we should plan when to stop supporting lua 5.1 and 5.2, and pick up this change after that?

**Comment by wilhelmy** [comment]
I agree it isn't very important for now.

Someone explained to me at some point that floats are just ints of [mantissa bits] length as long as the exponent part is zero, which is why Lua opted for double as the default type for their number type in previous releases, so as long as numbers stay below 2⁵³, which I guess they always will in notion's case, it doesn't make much of a practical difference.
